### PR TITLE
W-14104761 dynatrace log files gr

### PR DESCRIPTION
**Pull request by glenn-rodgers-sf**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

**Review by tbruno — DISMISSED** [review]
Wonder if we should also add some troubleshooting information

File: gateway/1.5/modules/ROOT/pages/_partials/task-config-logs-third-party.adoc [hunk 1]
```diff
@@ -34,22 +34,26 @@ As well as being able to configure additional Fluent Bit log output types, Flex
...
+Before configuring Flex Gateway log output, complete the following tasks:
 
 * xref:flex-install.adoc[Download Flex Gateway]
 * xref:flex-{page-mode}-reg-run.adoc[Register and Run Flex Gateway]
 
-Additionally, depending on your output, ensure that the service where you are directing logs to is operational. For Splunk, you must configure:
-
-* A running https://www.splunk.com/[Splunk^] service. You can test Splunk by setting up a free Splunk Cloud trial.
-* Added an HTTP Event Collector (HEC) data input to the Cloud or Enterprise Splunk service. See the https://docs.splunk.com/Documentation/Splunk/7.0.3/Data/UsetheHTTPEventCollector[Splunk documentation^] about how to add a data input, and how to create an HEC token.
+Additionally, depending on your output, ensure that the service where you are directing logs to is operational and that you have collected the necessary information:
```
Comment (Crispy-Salesforce) [inline gateway/1.5/modules/ROOT/pages/_partials/task-config-logs-third-party.adoc]: You can change the verb tense to present:
"Additionally, depending on your output, ensure that the service where you direct logs to is operational and that you have collected the necessary information:"

File: gateway/1.5/modules/ROOT/pages/_partials/task-config-logs-third-party.adoc [hunk 2]
```diff
@@ -34,22 +34,26 @@ As well as being able to configure additional Fluent Bit log output types, Flex
...
 
+* Splunk:
```
Comment (Crispy-Salesforce) [inline gateway/1.5/modules/ROOT/pages/_partials/task-config-logs-third-party.adoc:44]: I feel there is something missing in these bullets.
Is it possible to be consistent in their structure? Or is it possible to move this information to a table?
Something like this:
[options="header"]
|===
| Service | Necessary Information
| Splunk a| 
* A running link:https://www.splunk.com/[Splunk] service. You can test Splunk with a free Splunk Cloud trial.
* Added an HTTP Event Collector (HEC) data input to the Cloud or Enterprise Splunk service. See the link:https://docs.splunk.com/Documentation/Splunk/7.0.3/Data/UsetheHTTPEventCollector[Splunk documentation] about how to add a data input, and how to create an HEC token.
| Dynatrace a| 
* A running link:https://www.dynatrace.com/[Dynatrace] service. You can test Dynatrace with a free Dynatrace trial.
* Environment ID. To find your Dynatrace environment ID, see link:https://www.dynatrace.com/support/help/get-started/monitoring-environment/environment-id[Environment ID].
* Access token. To generate a Dynatrace access token, see link:https://www.dynatrace.com/support/help/manage/access-control/access-tokens[Access Tokens]. Ensure that the token has the `Ingest logs` token scope.
|===

File: gateway/1.5/modules/ROOT/pages/_partials/task-config-logs-third-party.adoc [hunk 3]
```diff
@@ -224,6 +232,51 @@ spec:
...
+      logLevel: info
+      outputs:
+        - dynatrace
+    accessLogs:
+      outputs:
+        - dynatrace
+
+----
+
+For more information about configuration options, see https://docs.fluentbit.io/manual/v/{fluentbit-url-ver-var}/pipeline/outputs/http[HTTP Output^]
```
Comment (Crispy-Salesforce) [inline gateway/1.5/modules/ROOT/pages/_partials/task-config-logs-third-party.adoc]: This sentence is missing its period.

**Review by Crispy-Salesforce — DISMISSED** [review]

File: gateway/1.5/modules/ROOT/pages/_partials/task-config-logs-third-party.adoc [hunk 4]
```diff
@@ -224,6 +236,51 @@ spec:
...
+        URI: /api/v2/logs/ingest
+        Format: json
+        json_date_key: timestamp
+        json_date_format: iso8601
+        tls: "on"
+        tls.verify: "off"
+    runtimeLogs:
+      logLevel: info
+      outputs:
+        - dynatrace
```
Comment (juanmolle) [inline gateway/1.5/modules/ROOT/pages/_partials/task-config-logs-third-party.adoc:273]: Do we have to add some comment regarding multiple Configuration Files? it happens to me that I have 2, one defining default output with runtimeLogs, and accessLogs, then the output is added to the final config but not the merge of these 2 elements. It take me some time to discover why

File: gateway/1.5/modules/ROOT/pages/_partials/task-config-logs-third-party.adoc [hunk 5]
```diff
@@ -34,22 +34,26 @@ As well as being able to configure additional Fluent Bit log output types, Flex
...
+Before configuring Flex Gateway log output, complete the following tasks:
 
 * xref:flex-install.adoc[Download Flex Gateway]
 * xref:flex-{page-mode}-reg-run.adoc[Register and Run Flex Gateway]
 
-Additionally, depending on your output, ensure that the service where you are directing logs to is operational. For Splunk, you must configure:
-
-* A running https://www.splunk.com/[Splunk^] service. You can test Splunk by setting up a free Splunk Cloud trial.
-* Added an HTTP Event Collector (HEC) data input to the Cloud or Enterprise Splunk service. See the https://docs.splunk.com/Documentation/Splunk/7.0.3/Data/UsetheHTTPEventCollector[Splunk documentation^] about how to add a data input, and how to create an HEC token.
+Additionally, depending on your output, ensure that the service where you direct logs to is operational and that you have collected the necessary information:
```
Comment (Crispy-Salesforce) [inline gateway/1.5/modules/ROOT/pages/_partials/task-config-logs-third-party.adoc]: I'd change the tense to:
"Additionally, depending on your output, ensure that the service where you direct logs to is operational and that you have the necessary information:"

File: gateway/1.5/modules/ROOT/pages/flex-troubleshoot-logging.adoc [hunk 6]
```diff
@@ -6,8 +6,32 @@ endif::[]
...
 
+* <<third-party-service, Logs Do Not Appear in a Third-Party Logging Service>>
 * <<incorrect-permissions,Logs Do Not Appear when Running in a VM>>
 
+[[third-party-service]]
+== Logs Do Not Appear in a Third-Party Logging Service
+
+After applying a logging configuration YAML file, logs should begin appearing within a few minutes. If your logs do not appear in your third party logging service:
+
+. Ensure that your third-party service is running. How you do this is dependent on your chosen service.
```
Comment (Crispy-Salesforce) [inline gateway/1.5/modules/ROOT/pages/flex-troubleshoot-logging.adoc]: I'd reword it to something like:
"Refer to your chosen service documentation to learn more about this topic."

File: gateway/1.5/modules/ROOT/pages/flex-troubleshoot-logging.adoc [hunk 7]
```diff
@@ -6,8 +6,32 @@ endif::[]
...
+. If the service is running, check the runtime logs for potential configuration errors.
++
+You can view the runtime logs for both Flex Gateways running in Connected Mode and Local Mode by following the steps in xref:flex-local-view-logs.adoc[]. Additionally, for Flex Gateways running in Connected Mode, you can xref:flex-view-logs-in-monitoring.adoc[view runtime logs in Anypoint Monitoring].
++
+Potential configuration errors may appear similar to the following:
```
Comment (Crispy-Salesforce) [inline gateway/1.5/modules/ROOT/pages/flex-troubleshoot-logging.adoc]: To avoid ambiguity, I'd say:
"Configuration errors can be similar to the following example:"
See ""can," "may," and "might"" in our Style Guide.

File: gateway/1.5/modules/ROOT/pages/flex-troubleshoot-logging.adoc [hunk 8]
```diff
@@ -6,8 +6,32 @@ endif::[]
...
++
+----
+[flex-gateway-fluent] [2023/09/07 20:17:49] [error] [output:http:http.1] hzk83039.live.dynatrace.com:443, HTTP status=401
+[flex-gateway-fluent] {"error":{"code":401,"message":"Missing authorization parameter."}}
+[flex-gateway-fluent] [2023/09/07 20:17:49] [ warn] [engine] failed to flush chunk '20-1694117863.683606251.flb', retry in 7 seconds: task_id=2, input=emitter_for_rewrite_tag.11 > output=http.1 (out_id=1)
+----
+
+. Regardless of if configuration errors appear in the runtime logs, view the `/var/tmp/mulesoft/flex-gateway/fluent-bit.conf` file to ensure your configuration values are correct.
```
Comment (Crispy-Salesforce) [inline gateway/1.5/modules/ROOT/pages/flex-troubleshoot-logging.adoc]: Maybe "whether" is a better option here:
"Whether or not configuration errors appear in the runtime logs, view the `/var/tmp/mulesoft/flex-gateway/fluent-bit.conf` file to ensure your configuration values are correct."

**Review by Crispy-Salesforce — APPROVED** [review]
